markovflow/models/variational_cvi_sde.py

```diff
@@ -0,0 +1,348 @@
+        """
+        Return the local objective and its gradients with regard to the expectation parameters.
+
+        :param Fmu: Means :math:`μ` with shape ``[..., latent_dim]``.
+        :param Fvar: Variances :math:`σ²` with shape ``[..., latent_dim]``.
+        :return: A local objective and gradient with regard to :math:`[μ, σ² + μ²]`.
+        """
+        with tf.GradientTape() as g:
+            g.watch([Fmu, Fvar])
+            local_obj = tf.reduce_sum(
+                input_tensor=self.local_objective(Fmu, Fvar, self._observations)
+            )
+        grads = g.gradient(local_obj, [Fmu, Fvar])
+        # turn into gradient wrt μ, σ² + μ²
+        grads = gradient_transformation_mean_var_to_expectation((Fmu, Fvar), grads)
+
+        return local_obj, grads
+
+    def cross_term(self, dist_q: StateSpaceModel = None) -> TensorType:
+        """
+        Calculate
+                    E_{q(x)[(f_q - f_L)\Sigma^-1(f_L - f_p)] .
+        """
+        # convert from state transitions of the SSM to SDE's drift and offset
+        if dist_q is None:
+            dist_q = self.dist_q
+
+        A_q = tf.squeeze(dist_q.state_transitions, axis=0)
+        b_q = tf.squeeze(dist_q.state_offsets, axis=0)
+        A_q = (A_q - tf.eye(self.state_dim, dtype=A_q.dtype)) / self.dt
+        b_q = b_q / self.dt
+
+        # convert from state transitions of the SSM to SDE's drift and offset
+        A_p_l = tf.squeeze(self.dist_p.state_transitions, axis=0)
+        b_p_l = tf.squeeze(self.dist_p.state_offsets, axis=0)
+        A_p_l = (A_p_l - tf.eye(self.state_dim, dtype=A_p_l.dtype)) / self.dt
+        b_p_l = b_p_l / self.dt
+
+        def func(x, t=None, A_q=A_q, b_q=b_q, A_p_l=A_p_l, b_p_l=b_p_l, sde_p=self.prior_sde):
+            # Adding N information
+            x = tf.transpose(x, perm=[1, 0, 2])
+            n_pnts = x.shape[1]
+
+            A_q = tf.repeat(A_q, n_pnts, axis=1)
+            b_q = tf.repeat(b_q, n_pnts, axis=1)
+            b_q = tf.expand_dims(b_q, axis=-1)
+
+            A_p_l = tf.repeat(A_p_l, n_pnts, axis=1)
+            b_p_l = tf.repeat(b_p_l, n_pnts, axis=1)
+            b_p_l = tf.expand_dims(b_p_l, axis=-1)
+
+            prior_drift = sde_p.drift(x=x, t=t)
+
+            fq_fL = ((x * A_q) + b_q) - ((x * A_p_l) + b_p_l)  # (f_q - f_L)
+            fl_fp = ((x * A_p_l) + b_p_l) - prior_drift  # (f_l - f_p)
+
+            sigma = sde_p.q
+            val = fq_fL * (1 / sigma) * fl_fp
+
+            return tf.transpose(val, perm=[1, 0, 2])
+
+        diag_quad = NDiagGHQuadrature(self.prior_sde.state_dim, 20)
+        q_mean = tf.squeeze(self.fx_mus, axis=0)[:-1]
+        q_covar = tf.squeeze(self.fx_covs, axis=0)[:-1]
+
+        val = diag_quad(func, q_mean, tf.squeeze(q_covar, axis=-1))
+        val = tf.reduce_sum(val) * self.dt
+
+        return val
+
+    def loss_lin(self, dist_p: StateSpaceModel = None, m: TensorType = None, S: TensorType = None):
```

Comment:
same, kl_linearization_loss ? 
and explain the name: the loss minimized for posterior linearization